src/main/java/com/atlassian/tutorial/rest/AttachmentsRestResource.java

```diff
@@ -0,0 +1,66 @@
+package com.atlassian.tutorial.rest;
+
+import com.atlassian.confluence.user.ConfluenceUser;
+import com.atlassian.tutorial.Service.AttachmentsService;
+import com.atlassian.tutorial.Service.AttachmentsServiceImpl;
+import org.apache.commons.codec.digest.HmacUtils;
+import org.apache.http.HttpStatus;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.security.SecureRandom;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.logging.Logger;
+
+import static com.atlassian.confluence.user.AuthenticatedUserThreadLocal.get;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+@Path("/attach")
+public class AttachmentsRestResource {
+    private static final Logger log = Logger.getLogger(AttachmentsRestResource.class.getName());
+    private final AttachmentsService attachmentsService;
+
+    public AttachmentsRestResource(AttachmentsService attachmentsService) {
+        this.attachmentsService = checkNotNull(attachmentsService);
+    }
+
+    @POST
+    @Produces({MediaType.APPLICATION_JSON})
+    @Path("/{pageID}")
+    public Response setAttachment(final @PathParam("pageID") String pageId, final @HeaderParam("path") String path) {
+        final ConfluenceUser confluenceUser = get();
```

Comment:
These two lines are used twice here and in the `getAttachmentPath()`. Remove code duplication please